app/src/backend/apiClient/deleteApiRecords.ts

```diff
@@ -0,0 +1,51 @@
+import firebaseApp from "../../firebase";
+import { doc, getFirestore, Timestamp, writeBatch } from "firebase/firestore";
+import { getOwnerId } from "backend/utils";
+import Logger from "lib/logger";
+
+export const deleteApiRecords = async (
+  uid: string,
+  recordIds: string[],
+  teamId?: string
+): Promise<{ success: boolean; data: unknown }> => {
+  if (!uid) {
+    return { success: false, data: null };
+  }
+
+  const ownerId = getOwnerId(uid, teamId);
+  const result = await deleteApiRecordsFromFirebase(uid, ownerId, recordIds);
+  return result;
+};
+
+const deleteApiRecordsFromFirebase = async (
+  uid: string,
+  ownerId: string,
+  recordIds: string[]
+): Promise<{ success: boolean; data: unknown }> => {
+  try {
+    const db = getFirestore(firebaseApp);
+    const batch = writeBatch(db);
+
+    recordIds.forEach((record) => {
+      const recordRef = doc(db, "apis", record);
+
+      batch.set(
+        recordRef,
+        {
+          deleted: true,
+          collectionId: "",
```

Comment:
Not any specific reason, just when deleted considering it as a single entity instead of having relation with other deleted entity (with may or may not exist)

Reply:
Keeping the collectionId might help in backtracking or recovering the records if deleted accidentally. wdyt?

Reply:
Yeah, make sense!